.github/workflows/master.yml

```diff
@@ -0,0 +1,88 @@
+name: Master
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  bump-version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.bump-version.outputs.tag }}
+    steps:
+      - name: Bump version and push tag
+        id: bump-version
+        uses: anothrNick/github-tag-action@1.17.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          WITH_V: true
+          DEFAULT_BUMP: patch
+  push-github-end2end:
+    runs-on: ubuntu-latest
+    needs: bump-version
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Push Docker Image to Github Registry
+        uses: whoan/docker-build-with-cache-action@v5
+        with:
+          username: "${{ github.actor }}"
+          password: "${{ secrets.GITHUB_TOKEN }}"
+          image_name: ${{ secrets.flytegithub_repo }}/flyteadmin
+          image_tag: latest,${{ github.sha }},${{ needs.bump-version.outputs.version }}
+          push_git_tag: true
+          registry: docker.pkg.github.com
+          build_extra_args: "--compress=true"
+      - uses: engineerd/setup-kind@v0.4.0
+      - name: End2End
```

Comment:
I think we should always remember that end-to-end tests should be smoke tests only - https://testing.googleblog.com/2015/04/just-say-no-to-more-end-to-end-tests.html#:~:text=While%20relying%20primarily%20on%20end,idea%20makes%20sense%20in%20theory.&text=Managers%20and%20decision%2Dmakers%20like,test%20would%20impact%20the%20user.

Reply:
Agreed..